File: Editor/src/main/java/org/turbomedia/turboedit/editor/renderer/FileResolveMethod.java
package org.turbomedia.turboedit.editor.renderer;

import static org.turbomedia.turboedit.editor.misc.Locale.GetText;

public enum FileResolveMethod {
    STREAMING("file_resolve_method.streaming"),
    MAPPING("file_resolve_method.mapping"),
    UPLOAD("file_resolve_method.upload");

    public final String translationKey;

    @Override
    public String toString() {
        return GetText(translationKey);
    }

    FileResolveMethod(String translationKey) {
        this.translationKey = translationKey;
    }
}

File: Editor/src/main/kotlin/turboedit/editor/renderer/FileResolveMethod.kt
package turboedit.editor.renderer

import turboedit.editor.misc.Locale

enum class FileResolveMethod(val translationKey: String) {
    STREAMING("file_resolve_method.streaming"),
    MAPPING("file_resolve_method.mapping"),
    UPLOAD("file_resolve_method.upload");

    override fun toString(): String {
        return Locale.getText(translationKey)
    }
}
